hypervisor/common/ptdev.c

```diff
@@ -40,28 +41,49 @@ static inline uint16_t ptirq_alloc_entry_id(void)
 	return (id < CONFIG_MAX_PT_IRQ_ENTRIES) ? id: INVALID_PTDEV_ENTRY_ID;
 }
 
+/*
+ * get the hash key when looking up ptirq_remapping_info from virtual
+ * source id in a vm, or just physical source id (vm == NULL).
+ */
+static inline uint64_t hash_key(const struct acrn_vm *vm,
+		const union source_id *sid)
+{
+	return hash64(sid->value + (uint64_t)vm, PTIRQ_ENTRY_HASHBITS);
```

Comment:
You change the hash algo ? If you add the "VM" to this hash key calculate, then you should optimize the find_ptirq_entry logic which should not check VM additionally.

Reply:
No. This is just an equivalent function to original calculation.